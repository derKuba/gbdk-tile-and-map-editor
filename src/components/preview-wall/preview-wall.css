:host {
  display: flex;
  width: 160px;
  flex-wrap: wrap;
  border: 1px solid;
  margin-top: 25px;
  margin-left: 50px;
  padding-top: 8px;
}
