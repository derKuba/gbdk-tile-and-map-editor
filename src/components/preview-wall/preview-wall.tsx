import { Component, Host, h } from '@stencil/core';

@Component({
    tag: 'mvjp-preview-wall',
    styleUrl: 'preview-wall.css',
    shadow: true,
})
export class PreviewWall {
    render() {
        let previewWall: HTMLElement[] = [];
        const FIELD_SIZE = 4;
        for (let index = 0; index < FIELD_SIZE; index++) {
            for (let j = 0; j < FIELD_SIZE; j++) {
                previewWall.push(<mvjp-preview size="s" standalone={true} />);
            }
        }

        return <Host>{previewWall}</Host>;
    }
}
