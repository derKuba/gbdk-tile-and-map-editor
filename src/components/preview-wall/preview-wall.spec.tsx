import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { PreviewWall } from './preview-wall';

describe('mvjp-preview-wall', () => {
    it('builds', () => {
        expect(new PreviewWall()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [PreviewWall],
            template: () => {
                return <mvjp-preview-wall />;
            },
        });

        expect(page.root).toMatchSnapshot();
    });
});
