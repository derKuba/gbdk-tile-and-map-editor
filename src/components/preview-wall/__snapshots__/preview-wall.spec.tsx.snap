// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-preview-wall matches snapshot 1`] = `
<mvjp-preview-wall>
  <mock:shadow-root>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
    <mvjp-preview size="s" standalone=""></mvjp-preview>
  </mock:shadow-root>
</mvjp-preview-wall>
`;
